src/expandable-section/expandable-section-header.tsx

```diff
@@ -6,26 +6,162 @@ import useFocusVisible from '../internal/hooks/focus-visible';
+import { generateUniqueId } from '../internal/hooks/use-unique-id';
 
-interface ExpandableSectionHeaderProps {
+interface ExpandableDefaultHeaderProps {
   id: string;
   className?: string;
-  variant: ExpandableSectionProps.Variant;
   children?: ReactNode;
   expanded: boolean;
   ariaControls: string;
-  ariaLabelledBy?: string;
   ariaLabel?: string;
   onKeyUp: KeyboardEventHandler;
   onKeyDown: KeyboardEventHandler;
   onClick: MouseEventHandler;
+  icon: JSX.Element;
+}
+
+interface ExpandableNavigationHeaderProps extends Omit<ExpandableDefaultHeaderProps, 'onKeyUp' | 'onKeyDown'> {
+  ariaLabelledBy?: string;
+}
+
+interface ExpandableContainerHeaderProps extends ExpandableDefaultHeaderProps {
+  headerDescription?: ReactNode;
+  headerCounter?: string;
+  headingTagOverride?: ExpandableSectionProps.HeadingTag;
+}
+
+interface ExpandableSectionHeaderProps extends Omit<ExpandableDefaultHeaderProps, 'children' | 'icon'> {
+  variant: ExpandableSectionProps.Variant;
+  header?: ReactNode;
+  headerText?: ReactNode;
+  headerDescription?: ReactNode;
+  headerCounter?: string;
+  headingTagOverride?: ExpandableSectionProps.HeadingTag;
+  ariaLabelledBy?: string;
 }
 
+const ExpandableDefaultHeader = ({
+  id,
+  className,
+  onClick,
+  ariaLabel,
+  ariaControls,
+  expanded,
+  children,
+  icon,
+  onKeyUp,
+  onKeyDown,
+}: ExpandableDefaultHeaderProps) => {
+  const focusVisible = useFocusVisible();
+  return (
+    <div
+      id={id}
+      role="button"
+      className={className}
+      tabIndex={0}
+      onKeyUp={onKeyUp}
+      onKeyDown={onKeyDown}
+      onClick={onClick}
+      aria-label={ariaLabel}
+      aria-controls={ariaControls}
+      aria-expanded={expanded}
+      {...focusVisible}
+    >
+      <div className={styles['icon-container']}>{icon}</div>
+      {children}
+    </div>
+  );
+};
+
+const ExpandableNavigationHeader = ({
+  id,
+  className,
+  onClick,
+  ariaLabelledBy,
+  ariaLabel,
+  ariaControls,
+  expanded,
+  children,
+  icon,
+}: ExpandableNavigationHeaderProps) => {
+  const focusVisible = useFocusVisible();
+  return (
+    <div id={id} className={className} onClick={onClick}>
+      <button
+        className={styles['icon-container']}
+        aria-labelledby={ariaLabelledBy}
+        aria-label={ariaLabel}
+        aria-controls={ariaControls}
+        aria-expanded={expanded}
+        {...focusVisible}
+      >
+        {icon}
+      </button>
+      {children}
+    </div>
+  );
+};
+
+const ExpandableContainerHeader = ({
+  id,
+  className,
+  onClick,
+  ariaLabel,
+  ariaControls,
+  expanded,
+  children,
+  icon,
+  headerDescription,
+  headerCounter,
+  headingTagOverride,
+  onKeyUp,
+  onKeyDown,
+}: ExpandableContainerHeaderProps) => {
+  const focusVisible = useFocusVisible();
+  const screenreaderContentId = generateUniqueId('expandable-section-header-content-');
+  return (
+    <div id={id} className={className} onClick={onClick} {...focusVisible}>
```

Comment:
I am thinking if we can probably add a dev warning if noticed more than one focused element under the header, wdyt?

```
const ref = useRef(null)

useEffect(() => {
  if (ref.current) {
    const focusables = getFocusables(ref.current)
    if (focusables.length > 1) {
      warnOnce('ExpandableSection', 'Do not put focusable elements...')
    }
  }
}, [])

return <div ref={ref} id={id} ...>...</div>
```

That is probably relevant for all header variants yet for container it also leads to the additional outline being rendered.

Reply:
I kinda think this is one of the common misuses customers may have. Adding dev warn for such case might end up a long list dev warn, and distract customers from more important warnings?

Reply:
The dev warning is only added once per component so I don't think it will be distracting, but updating the guidelines might be enough - I dont't have a strong opinion on whether we should add the warning.